src/QtAVPlayer/qavplayer.cpp

```diff
@@ -329,9 +329,13 @@ void QAVPlayerPrivate::terminate()
     if (dev)
         dev->abort(true);
     loaderFuture.waitForFinished();
-    demuxerFuture.waitForFinished();
     videoPlayFuture.waitForFinished();
     audioPlayFuture.waitForFinished();
+
+    demuxer.abort();
+    demuxerFuture.waitForFinished();
+    demuxer.abort(false);
```

Comment:
interesting why second call is needed there?

Reply:
I have added it to "reset" to his previous state (since abort is `false` by default). Thus, other methods from FFmpeg will not be prematurely aborted (thinking about TEARDOWN signals not being sent for example).

Reply:
oh, forgot that can reset the flag. As I see it is reset already in `onLoad`, so should work without this line.